src/generate/rust.rs

```diff
@@ -944,7 +924,7 @@ where
         };
         quote!(
             #[allow(non_camel_case_types)]
-            pub struct #ident<'a, 'i: 'a, I: 'a + ::gll::runtime::Input> {
+            pub struct #ident<'a, 'i: 'a, I: crate::gll::runtime::Input> {
```

Comment:
We can get rid of `'i: 'a` too, right? Can we do some of these in a separate PR, before the move to 2018?

Reply:
Yeah that worked.